### ISSUE TITLE
[Feature Request] Option to drag viewpoint by holding middle-button instead of right-button

### ISSUE THREAD

**Issue by ot5darva** [OpenRCT2/OpenRCT2#5388]
OS: Windows 8.1
Version: 0.0.7

To prevent accidentally deleting/editing objects while moving the viewport by mouse dragging, I think adding an option to use the middle mouse button for dragging of the viewport would be very useful. Holding down the middle mouse button for scrolling or panning is a common convention in many programs so I think this new option will be a good addition. Enabling this option should also disable using the right button for viewport dragging.

Sample on possible location to access the option and wording:
![options sample middle mouse view port dragging](https://cloud.githubusercontent.com/assets/26368692/25115124/b5f11f6a-23b8-11e7-8650-e141a1f90f54.PNG)

**Comment by harrysbird** [comment]
I Love the idea.

I thick it Would Requires new save format or not, we don't know when start developing new save format.

**Comment by pizza2004** [comment]
This has absolutely nothing to do with the save format and could easily be implemented now assuming it would even be wanted.  I've seen a lot of talk about keeping the options menu as simple as possible.  If this went anywhere I'd suggest just allowing the user to select which mouse button they want to do what under the shortcut keys menu.

**Comment by Gymnasiast** [comment]
@harrysbird It does not require a new save format.

**Comment by mrtnptrs** [comment]
@harrysbird not everything needs a new save format (just like I told you here: https://github.com/OpenRCT2/OpenRCT2/issues/5340). These are the last two comments you made here at the issue tracker (you made some other, more usefull comments at two other closed issues earlier).

If a feature request requires a new save format, the developers will tell the person who opened the issue. It could be possible that you confuse others with this. So, let the developers handle it if something is possible or not. 

Also, think before you say something: why the hell would a window dragging option require a new save format?! It will require a new config option (involving the config file of course.) So, it has nothing to do with the save file. :)

Oh, "**we** don't know when **we** start developing new save format". Lol, hey pal, are you an OpenRCT2 developer now? :P 

I fully agree with @pizza2004 his opinion on this btw.

**Comment by Gymnasiast** [comment]
Note: I got rid of several off-topic comments.

**Comment by xp4xbox** [comment]
It would be nice to add it to the shortcut menu, since some of us do not have middle buttons.

**Comment by marijnvdwerf** [comment]
What other programs use the centre mouse button for dragging?

**Comment by Gymnasiast** [comment]
To play the devil's advocate: it's already possible to shift the viewport by holding the pointer to the screen edge, or using the keyboard. Is it really necessary to add this option?

**Comment by ot5darva** [comment]
@xp4xbox I don't agree with having this option in the shortcut menu because everything in that menu pertains to keyboard inputs.

@marijnvdwerf Every browser, every program in Microsoft office, and also in most 3D graphics software like Blender. In Blender, holding the middle mouse button rotates but holding shift while also holding middle button pans the camera.

@Gymnasiast I play the game while in windowed mode because it allows me to look at reference pictures and do other things while I'm building. I assume many others do that too. Therefore I don't move the viewport by having the pointer at the edge of the screen. Also, using this method to move the viewport is very slow. Using the directional keys also moves at this same slow speed. It is already possible to move the viewport while holding the right mouse button but since the right mouse button also deletes/edits objects, I feel having the option to change it to middle click helps prevent that while keeping the usability of moving around with the mouse. Also many programs use the middle mouse for scrolling so I think this addition is fitting to help this game behave similarly to modern programs. I hope you will reconsider the viability of adding this option. I think many people will benefit from it.

**Comment by rwjuk** [comment]
@ot5darva In those programs you mention, it's not 'click-drag the middle button', it's 'click the button, then move the cursor'. Not in any way saying we should replicate that, but click-drag with middle mouse isn't exactly a common convention.

**Comment by ot5darva** [comment]
@rwjuk It's both the way you described and also the way I described. You can click on the middle button to engage scroll mode and click again to disengage. You can also hold down the middle button to drag scroll around and release to stop the function.

**Comment by rwjuk** [comment]
@ot5darva Hmm, you're right - I forgot it did that.

Personally I think we should add it & stick the option in the shortcut menu. I entirely agree that fewer options makes for a better user experience, but I can see this being genuinely useful.

**Comment by SevenChords** [comment]
any update on this? i ask because when i delete a row of paths, the game often decides to scroll because i am somehow holding down the right mouse button too long when right clicking to delete path, i'd really love to change that to another key/mouse button or remove it entirely for my own controls (i scroll with the keys anyway)

**Comment by xp4xbox** [comment]
Might look into this.
